src/base/io/utilities/readSBML.m

```diff
@@ -1,787 +1,494 @@
+    %get the parameters (needed for fluxes), we can extract the data from
+    %the fluxes.
+    parameterids = {modelSBML.parameter.id};
+    paramvalues = [modelSBML.parameter.value];
+    lbvals = {sbmlReactions.fbc_lowerFluxBound};
+    ubvals = {sbmlReactions.fbc_upperFluxBound};
+    [pres,pos] = ismember(ubvals,parameterids);
+    model.ub = defaultBound*ones(numel(model.rxns),1);
+    model.ub(pres) = double(paramvalues(pos(pres))); %Conversion to double
+    model.lb = -defaultBound*ones(numel(model.rxns),1).*double([sbmlReactions.reversible]');
+    [pres,pos] = ismember(lbvals,parameterids);
+    model.lb(pres) = double(paramvalues(pos(pres)));%Conversion to double
+else
+    %if there is a kineticLaw field, we might have old cobra style
+    %annotations.
+    if isfield(sbmlReactions,'kineticLaw')
+        kineticLaws = {sbmlReactions.kineticLaw};
+        %if there are none use defautlts.
+        if all(cellfun(@isempty, kineticLaws))
+            model.ub = defaultBound*ones(numel(model.rxns),1);
+            model.lb = -defaultBound*ones(numel(model.rxns),1).*double([sbmlReactions.reversible]');
+            model.c = zeros(numel(model.rxns),1);
         else
-            ub(i) = defaultBound;
-            if (rev(i) == 1)
-                lb(i) = -defaultBound;
-            else
-                lb(i) = 0;
-            end
+            %or parse the annotations.
+            [lb,ub,obj] = cellfun(@(x,y) extractBounds(x,defaultBound,y),kineticLaws, num2cell([sbmlReactions.reversible]));
+            model.lb = columnVector(lb);
+            model.ub = columnVector(ub);
+            model.c = columnVector(obj);
         end
+    else
+        model.ub = defaultBound*ones(numel(model.rxns),1);
+        model.lb = -defaultBound*ones(numel(model.rxns),1).*double([sbmlReactions.reversible]');
+        model.c = zeros(numel(model.rxns),1);
     end
-
 end
 
-%% gene
-
 
-warning off
-%% Construct gene to rxn mapping
-
-if isfield(modelSBML,'fbc_version')&&modelSBML.fbc_version==2   % in the case of the fbc v2 file, the gene products are stored in the different XML attributes
-    allGenes = {};
-    for i=1:size(modelSBML.fbc_geneProduct,2)
-        allGenes{i,1}=modelSBML.fbc_geneProduct(i).fbc_label; % according to Recon2 COBRA structure, 'allGenes' are converted to "model.genes".
-    end
-end
 
-allGenes = unique(allGenes);
-
-if (hasNotesField)||(isfield(modelSBML,'fbc_version')&&(modelSBML.fbc_version==2))
-    rxnGeneMat = sparse(nRxns,length(allGenes));
-    for i = 1:nRxns
-        if iscell(genes{i})
-            [tmp,geneInd] = ismember(genes{i},allGenes);
+%% Set up the objective. The default is maximisation.
+model.osense = -1;
+if isfield(modelSBML,'fbc_objective')
+    %We only support the first one we find
+    if ~isempty(modelSBML.fbc_objective)
+        osenseStr = modelSBML.fbc_objective(1).fbc_type;
+        if strcmp(osenseStr , 'maximize')
+            model.osense = -1;
```

Comment:
This condition is not necessary, as `model.osense` has already been initialized to `-1`.